### PR TITLE
Rename template test integration

### PR DESCRIPTION
**Pull request by janboll**
Rename it to resource template test, cause it's testing resource templates. Template Test would collide with future addition of templating

[APPSRE-8710](https://issues.redhat.com/browse/APPSRE-8710)

### CODE REVIEW

File: reconcile/cli.py [hunk 1]
```diff
@@ -2628,10 +2628,10 @@ def prometheus_rules_tester(ctx, thread_pool_size, cluster_name):
 
 @integration.command(short_help="Tests templating of resources.")
 @click.pass_context
-def template_tester(ctx):
-    import reconcile.template_tester
+def resource_template_tester(ctx):
```
Comment (hemslo) [inline reconcile/cli.py:2631]: Rename here is a breaking change, need to have additional promotion notes.

File: reconcile/resource_template_tester.py [hunk 2]
```diff
@@ -11,13 +11,13 @@
...
 
-QONTRACT_INTEGRATION = "template-tester"
+QONTRACT_INTEGRATION = "resource-template-tester"
 QONTRACT_INTEGRATION_VERSION = make_semver(0, 1, 0)
 
 
 TEMPLATE_TESTS_QUERY = """
 {
-  tests: template_tests_v1 {
+  tests: resource_template_tests_v1 {
```
Comment (hemslo) [inline reconcile/resource_template_tester.py:20]: missing gql schema sync?

Comment (janboll) [reply reconcile/resource_template_tester.py:20]: Yes, it's an old-school integration.

File: reconcile/cli.py (hunk 1 above)
Comment (janboll) [reply reconcile/cli.py:2631]: For fedramp? Will create one

File: reconcile/resource_template_tester.py (hunk 2 above)
Comment (hemslo) [reply reconcile/resource_template_tester.py:20]: I mean there is no `make gql-introspection` change for new schema?

Comment (janboll) [reply reconcile/resource_template_tester.py:20]: Oh, there is. it's not required since that validation is not used here. I can add it.

**Review by hemslo — APPROVED** [review]
LGTM!

**Pull request merged**